calico_rkt/calico_rkt.py

```diff
@@ -14,387 +14,485 @@
+
+        :return: None.
+        """
+        _log.info('Configuring pod %s' % self.container_id)
+
+        # Step 1: Assign an IP address using the given IPAM plugin.
+        assigned_ip = self._assign_ip()
+
+        # Step 2: Create the Calico endpoint object.
+        endpoint = self._create_endpoint(assigned_ip)
+
+        # Step 3: Provision the veth for this endpoint.
+        endpoint = self._provision_veth(endpoint)
+        
+        # Step 4: Provision / set profile on the created endpoint.
+        self._set_profile_on_endpoint(endpoint, self.network_name)
+
+        # Step 5: If all successful, print the IPAM plugin's output to stdout.
+        dump = json.dumps(self.ipam_result)
+        _log.info("Printing CNI result to stdout: %s", dump)
+        print(dump)
+
+        _log.info("Finished creating pod %s", self.container_id)
+    
+    def delete(self):
+        """Handles CNI_CMD_DELETE requests.
+
+        Remove this pod from Calico networking.
+
+        :return: None.
+        """
+        _log.info('Deleting pod %s' % self.container_id)
+
+        # Step 1: Remove any IP assignments.
+        self._release_ip()
+
+        # Step 2: Get the Calico endpoint for this workload. If it does not
+        # exist, log a warning and exit successfully.
+        endpoint = self._get_endpoint()
+        if not endpoint:
+            _log.info("No more to do since endpoint does not exist.",
+                       self.container_id)
+            sys.exit(0)
+
+        # Step 3: Delete the veth interface for this endpoint.
+        _log.info("Removing veth for endpoint %s", endpoint.name)
+        netns.remove_veth(endpoint.name)
+
+        # Step 4: Delete the Calico endpoint.
+        self._remove_endpoint()
+
+        # Step 5: Delete any profiles for this endpoint
+        # TODO: This is racey and needs further thinking.
+        self._remove_profile(self.network_name)
+
+        _log.info('Finished deleting pod %s' % self.container_id)
+
+    def _assign_ip(self):
+        """Assigns and returns an IPv4 address using the IPAM plugin
+        specified in the network config file.
+
+        :return: IPAddress - The assigned IP address.
+        """
+        # TODO - Handle errors thrown by IPAM plugin
+        self.ipam_result = self._call_ipam_plugin()
+        _log.debug("IPAM plugin result: %s", self.ipam_result)
 
+        try:
+            # Load the response and get the assigned IP address.
+            self.ipam_result = json.loads(self.ipam_result)
+        except ValueError:
+            _log.exception("Failed to parse IPAM response, exiting")
+            sys.exit(1)
 
-def delete(args):
-    """
-    Cleanup after a pod.
+        try:
+            assigned_ip = IPAddress(self.ipam_result["ip4"]["ip"])
+        except KeyError:
+            _log.error("IPAM plugin did not return an IPv4 address.")
+            sys.exit(1)
+        except (AddrFormatError, ValueError):
```

Comment:
Yes it throws both - throws a ValueError if there is a '/' in the ip addr string; throws AddrFormatError for all other format errors.